**HTTP-mode verification — confirming the fix preserves the empty-secret guard**

Tested the patched build in HTTP mode (`socket_mode_enabled: false`) with **no `SLACK_SIGNING_SECRET` set**. As intended, the request now reaches the middleware and the verifier raises lazily on first use — HTTP requests still require a non-empty signing secret (the behavior the new `test_http_request_with_empty_signing_secret_raises` tests lock in), so an unconfigured secret surfaces as a clear error rather than being silently accepted:

```
ValueError: signing_secret must not be empty.
127.0.0.1 - - [REDACTED] "POST /slack/events HTTP/1.1" 500 -
DEBUG:slack_bolt.App:Applying slack_bolt.middleware.ssl_check.ssl_check.SslCheck
DEBUG:slack_bolt.App:Applying slack_bolt.middleware.request_verification.request_verification.RequestVerification
ERROR:slack_bolt.App:Failed to run a middleware (error: signing_secret must not be empty.)
Traceback (most recent call last):
  File ".../slack_bolt/app/app.py", line 561, in dispatch
    resp = middleware.process(req=req, resp=resp, next=middleware_next)
  File ".../slack_bolt/middleware/request_verification/request_verification.py", line 51, in process
    if self.verifier.is_valid(body, timestamp, signature):
       ^^^^^^^^^^^^^
  File ".../slack_bolt/middleware/request_verification/request_verification.py", line 32, in verifier
    self._verifier = SignatureVerifier(signing_secret=self._signing_secret)
  File ".../slack_sdk/signature/__init__.py", line 29, in __init__
    self.signing_secret = signing_secret
  File ".../slack_sdk/signature/__init__.py", line 41, in signing_secret
    raise ValueError("signing_secret must not be empty.")
```

For Socket Mode (the original #1535 report), the verifier is never constructed and the app starts cleanly. ✅

--- a/slack_bolt/middleware/request_verification/request_verification.py
+++ b/slack_bolt/middleware/request_verification/request_verification.py
@@ -26,8 +26,7 @@ class RequestVerification(Middleware):
 
     @property
     def verifier(self) -> SignatureVerifier:
-        # Defer initialization to avoid errors of a missing signing secret for
-        # apps using Socket Mode connections
+        # Defer initialization to avoid errors during start up
         if self._verifier is None:
             self._verifier = SignatureVerifier(signing_secret=self._signing_secret)
         return self._verifier

